apis/config/v1beta1/defaults.go

```diff
@@ -87,8 +87,14 @@ func SetDefaults_Configuration(cfg *Configuration) {
 	if cfg.ClientConnection.Burst == nil {
 		cfg.ClientConnection.Burst = pointer.Int32(DefaultClientConnectionBurst)
 	}
-	if cfg.WaitForPodsReady != nil && cfg.WaitForPodsReady.Timeout == nil {
-		cfg.WaitForPodsReady.Timeout = &metav1.Duration{Duration: defaultPodsReadyTimeout}
+	if cfg.WaitForPodsReady != nil {
+		if cfg.WaitForPodsReady.Timeout == nil {
+			cfg.WaitForPodsReady.Timeout = &metav1.Duration{Duration: defaultPodsReadyTimeout}
+		}
+		if cfg.WaitForPodsReady.BlockAdmission == nil {
+			defaultBlockAdmission := true
+			cfg.WaitForPodsReady.BlockAdmission = &defaultBlockAdmission
```

Comment:
In this implementation, wouldn't we enforce a `cfg.WaitForPodsReady.BlockAdmission=true` even if a `cfg.WaitForPodsReady.Enable` is false?

Reply:
yes, but`cfg.WaitForPodsReady.BlockAdmission` won't work if `cfg.WaitForPodsReady.Enable` is false. 
I can add some checking to set it false when `cfg.WaitForPodsReady.Enable` is false if needed

Reply:
IIRC, a config after defaulting is shown in the kueue-controller log. 
So, I think that users would confuse a bit with a `cfg.WaitForPodsReady.BlockAdmission=true` and a `cfg.WaitForPodsReady.Enable=false`.

Hence, if a `cfg.WaitForPodsReady.Enable` is false, not defaulting a `cfg.WaitForPodsReady.BlockAdmission` to true would be good.

cc: @mimowo

Reply:
Not defaulting when `.Enable=false` sounds good to me. But it is also ok for me to leave defaulting as true if the feature is disabled, to keep it simple (as we default timeout unconditionally).